core/grants.go

```diff
@@ -0,0 +1,187 @@
+package core
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+
+	"chain/errors"
+	"chain/net/http/authz"
+	"chain/net/http/httpjson"
+)
+
+// an api-friendly representation of a grant
+type apiGrant struct {
+	GuardType string                 `json:"guard_type"`
+	GuardData map[string]interface{} `json:"guard_data"`
+	Policy    string                 `json:"policy"`
+	CreatedAt string                 `json:"created_at"`
+}
+
+// ErrMissingTokenID is returned when a token does not exist.
+var errMissingTokenID = errors.New("id does not exist")
+
+func (a *API) createGrant(ctx context.Context, x apiGrant) error {
+	id, ok := x.GuardData["id"]
+	if ok {
+		if strID, ok := id.(string); !ok || !a.accessTokens.Exists(ctx, strID) {
+			return errMissingTokenID
+		}
+	} else {
+		return errMissingTokenID
+	}
+
+	guardData, err := json.Marshal(x.GuardData)
```

Comment:
I think it would be good to leave a note here that we rely on the consistent serialization format to compare grants for equality.

```
// NOTE: package json produces consistent serialization output,
// effectively an ad hoc canonical form. We rely on this to
// grant data for equality.
```